frontend/src/views/Lobby.tsx

```diff
@@ -122,11 +125,17 @@ function LobbyPage() {
   useEffect(() => {
     // Grab the user and room information; otherwise, redirect to the join page
     if (checkLocationState(location, 'user', 'roomId')) {
-      setCurrentUser(location.state.user);
-
       // Call GET endpoint to get latest room info
       getRoom(location.state.roomId)
-        .then((res) => setStateFromRoom(res))
+        .then((res) => {
+          setStateFromRoom(res);
+          // Reset the user to hold the ID.
+          res.inactiveUsers.forEach((user: User) => {
+            if (user.nickname === location.state.user.nickname) {
+              setCurrentUser(user);
+            }
+          });
```

Comment:
Yeah I think since we have to make this REST call anyway to get all the room details anyway, I'm fine having this here; however, let me know if you think I should still move this.